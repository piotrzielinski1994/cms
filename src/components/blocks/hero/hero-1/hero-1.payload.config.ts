import { AdminTranslations } from '@/payload/locale';
import type { Block } from 'payload';

export const hero1BlockPayloadConfig = {
  slug: 'hero-1',
  interfaceName: 'Hero1Block',
  fields: [
    {
      name: 'heading',
      type: 'text',
      localized: true,
      label: ({ t }: { t: AdminTranslations }) => t('fields:heading'),
    },
    {
      name: 'subheading',
      type: 'text',
      required: true,
      localized: true,
      label: ({ t }: { t: AdminTranslations }) => t('fields:subheading'),
    },
    {
      name: 'buttons',
      type: 'array',
      label: ({ t }: { t: AdminTranslations }) => t('fields:buttons'),
      admin: {
        initCollapsed: true,
      },
      fields: [
        {
          name: 'label',
          type: 'text',
          required: true,
          localized: true,
          label: ({ t }: { t: AdminTranslations }) => t('fields:label'),
        },
        {
          name: 'reference',
          type: 'relationship',
          label: ({ t }: { t: AdminTranslations }) => t('fields:documentToLinkTo'),
          relationTo: ['pages', 'posts'],
          required: true,
        },
        {
          name: 'selector',
          type: 'text',
          label: ({ t }: { t: AdminTranslations }) => t('fields:selector'),
        },
      ],
    },
    // {
    //   name: 'cta',
    //   type: 'group',
    //   label: ({ t }: { t: AdminTranslations }) => t('fields:cta'),
    //   defaultValue: undefined,
    //   fields: [
    //     {
    //       name: 'label',
    //       type: 'text',
    //       // TODO: required: true,
    //       localized: true,
    //       label: ({ t }: { t: AdminTranslations }) => t('fields:label'),
    //     },
    //     {
    //       name: 'reference',
    //       type: 'relationship',
    //       label: ({ t }: { t: AdminTranslations }) => t('fields:documentToLinkTo'),
    //       relationTo: ['pages', 'posts'],
    //       // TODO: required: true,
    //     },
    //     {
    //       name: 'selector',
    //       type: 'text',
    //       label: ({ t }: { t: AdminTranslations }) => t('fields:selector'),
    //     },
    //   ],
    // },
  ],
} satisfies Block;
