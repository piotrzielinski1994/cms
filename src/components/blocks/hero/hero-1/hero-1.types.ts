import { Hero1Block } from '@/payload.types';

export interface Hero1Props extends Hero1Block {}
