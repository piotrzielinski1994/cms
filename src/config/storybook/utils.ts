import { ReactNode } from 'react';

const DEFAULT_VALUE = '__default__' as const;
const THUMBNAIL_ID = 'thumbnail';

const storyToUrlPath = (title: string) => {
  const parts = title.split('/');
  const story = parts[parts.length - 1];
  const titlePart = title.toLowerCase().replace(/\//g, '-').replace(/\s+/g, '-');
  const storyPart = story
    .replace(/([a-z0-9])([A-Z])/g, '$1-$2')
    .replace(/([a-zA-Z])(\d+)/g, '$1-$2')
    .toLowerCase();
  return `${titlePart}--${storyPart}`;
};

const getFallback = <T extends ReactNode>(value: T, translation: string): T => {
  return value === DEFAULT_VALUE ? (translation as T) : value;
};

export { DEFAULT_VALUE, getFallback, storyToUrlPath, THUMBNAIL_ID };
