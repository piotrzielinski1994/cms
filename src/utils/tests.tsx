import { Providers } from '@/providers';
import { ComponentProps, ReactNode } from 'react';

const resolveServerComponent = async <T extends (props: unknown) => Promise<ReactNode>>(
  Component: T,
  props: Parameters<T>[0],
) => {
  const ComponentResolved = await Component(props);
  return () => ComponentResolved;
};

const withProviders = (
  children: ReactNode,
  props: Partial<Omit<ComponentProps<typeof Providers>, 'children'>> = {},
) => {
  const overwrittenProps: ComponentProps<typeof Providers> = {
    locale: 'en',
    initialTheme: 'light',
    initialFontScale: 'base',
    initialCookiesConsent: true,
    ...props,
    children,
  };
  return <Providers {...overwrittenProps} />;
};

export { resolveServerComponent, withProviders };
