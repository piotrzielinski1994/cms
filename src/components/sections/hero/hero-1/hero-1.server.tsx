import { Hero1Block, Page } from '@/payload/payload.types';
import { Hero1 as Hero1Component } from './hero-1';

const Hero1 = ({ heading, subheading, buttons = [] }: Hero1Block) => {
  return (
    <Hero1Component
      heading={heading ?? undefined}
      subheading={subheading ?? undefined}
      buttons={buttons?.map((button, index) => {
        const href = (button?.reference?.value as Page).path ?? '';
        const variant = index === 0 ? 'primary' : 'secondary';
        return { label: button.label, href, variant };
      })}
    />
  );
};

export { Hero1 };
