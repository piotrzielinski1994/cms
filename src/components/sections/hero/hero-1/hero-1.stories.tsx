import type { Meta, StoryObj } from '@storybook/react';
import { type ComponentProps } from 'react';
import { Hero1 as Hero1Component } from './hero-1';

type Args = ComponentProps<typeof Hero1Component>;

const meta: Meta<Args> = {
  component: Hero1Component,
  title: 'Components/Sections/Hero1',
  argTypes: {
    heading: { control: 'text' },
    subheading: { control: 'text' },
    buttons: { control: 'object' },
  },
  args: {
    heading: 'Heading',
    subheading: 'Subheading',
    buttons: [
      { label: 'Button 1', href: '/path-1', variant: 'primary' },
      { label: 'Button 2', href: '/path-2', variant: 'secondary' },
    ],
  },
};

const Render = (args: Args) => {
  return <Hero1Component {...args} />;
};

const Hero1: StoryObj<typeof Hero1Component> = { render: Render };

export { Hero1 };
export default meta;
