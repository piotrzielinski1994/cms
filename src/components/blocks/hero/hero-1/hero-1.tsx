import { ButtonLink } from '@/components/basic/button';
import { ContactForm } from '@/components/blocks/form/contact-form/contact-form';
import { Container } from '@/components/layout/container/container';
import { Section } from '@/components/layout/section/section';
import { Hero1Block, Page } from '@/payload/payload.types';
import { cn } from '@/utils/tailwind';

const Hero1 = ({ heading, subheading, buttons }: Hero1Block) => {
  return (
    <Section>
      <Container
        className={cn('py-10', 'justify-items-start content-center gap-4', 'bg-background1')}
      >
        <h1 className={cn('text-6xl font-semibold')}>{heading}</h1>
        <ContactForm />
        {subheading && <p>{subheading}</p>}
        {buttons?.map((button) => {
          const path = (button?.reference?.value as Page).path;
          return (
            <ButtonLink
              key={button.label}
              href={`${path}${button.selector ? '#' + button.selector : ''}`}
            >
              {button.label}
            </ButtonLink>
          );
        })}
      </Container>
    </Section>
  );
};

export { Hero1 };
