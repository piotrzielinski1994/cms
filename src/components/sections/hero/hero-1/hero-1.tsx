import { ButtonLink } from '@/components/basic/button/button';
import { Container } from '@/components/basic/container';
import { Section } from '@/components/basic/section';
import { cn } from '@/utils/tailwind';
import { ComponentProps } from 'react';

type Hero1Props = {
  heading?: string;
  subheading?: string;
  buttons?: Array<Pick<ComponentProps<typeof ButtonLink>, 'href' | 'variant'> & { label: string }>;
};

const Hero1 = ({ heading, subheading, buttons = [] }: Hero1Props) => {
  return (
    <Section>
      <Container
        className={cn('py-10', 'justify-items-start content-center gap-4', 'bg-background1')}
      >
        <h1 className={cn('text-6xl font-semibold')}>{heading}</h1>
        {subheading && <p>{subheading}</p>}
        {buttons.length > 0 && (
          <div className="flex gap-4">
            {buttons.map((button) => {
              return (
                <ButtonLink key={button.label} href={button.href} variant={button.variant}>
                  {button.label}
                </ButtonLink>
              );
            })}
          </div>
        )}
      </Container>
    </Section>
  );
};

export { Hero1 };
