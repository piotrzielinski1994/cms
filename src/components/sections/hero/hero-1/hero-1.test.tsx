import { withProviders } from '@/utils/tests';
import { render, waitFor } from '@testing-library/react';
import { ComponentProps } from 'react';
import { describe, expect, it } from 'vitest';
import { axe } from 'vitest-axe';
import { Hero1 } from './hero-1';

describe('Hero1', () => {
  const defaultProps = {
    heading: 'Heading',
    subheading: 'Subheading',
    buttons: [
      { label: 'Button 1', href: '/path-1', variant: 'primary' },
      { label: 'Button 2', href: '/path-2', variant: 'secondary' },
    ],
  } satisfies ComponentProps<typeof Hero1>;

  it('should have no accessibility violations', async () => {
    const { container } = await waitFor(() => render(withProviders(<Hero1 {...defaultProps} />)));
    const results = await axe(container);
    expect(results).toHaveNoViolations();
  });

  it('should match the snapshot', async () => {
    const { container } = render(withProviders(<Hero1 {...defaultProps} />));
    expect(container).toMatchSnapshot();
  });
});
