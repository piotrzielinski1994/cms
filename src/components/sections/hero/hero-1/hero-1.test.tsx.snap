// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Hero1 > should match the snapshot 1`] = `
<div>
  <section
    class="px-4 sm:px-6 grid cms-section"
  >
    <div
      class="w-full max-w-screen-2xl mx-auto grid cms-container py-10 justify-items-start content-center gap-4 bg-background1"
    >
      <h1
        class="text-6xl font-semibold"
      >
        Heading
      </h1>
      <p>
        Subheading
      </p>
      <div
        class="flex gap-4"
      >
        <a
          class="px-6 py-3 border-2 border-primary hover:border-primary/90 font-semibold whitespace-nowrap disabled:cursor-not-allowed bg-primary hover:bg-primary/90 text-primary-foreground disabled:border-primary/0 disabled:bg-primary/50"
          href="/path-1"
        >
          Button 1
        </a>
        <a
          class="px-6 py-3 border-2 border-primary hover:border-primary/90 font-semibold whitespace-nowrap disabled:cursor-not-allowed text-primary disabled:border-primary/50 disabled:text-primary/50 bg-background [&:not(:disabled)]:hover:bg-foreground/5"
          href="/path-2"
        >
          Button 2
        </a>
      </div>
    </div>
  </section>
</div>
`;
